cli/injector/main.go

```diff
@@ -354,6 +360,18 @@ func cleanAndExit(cmd *cobra.Command, args []string) {
 		log.Fatalw(fmt.Sprintf("disruption cleanup failed on %d injectors (comma separated errors)", len(errs)), "errors", combined.String())
 	}
 
+	pod, err := configs[0].K8sClient.CoreV1().Pods(chaosNamespace).Get(context.Background(), os.Getenv(env.Hostname), metav1.GetOptions{})
+	if err != nil {
+		log.Errorw("couldn't GET this pod in order to remove its finalizer", "pod", os.Getenv(env.Hostname), "err", err)
```

Comment:
Isn't it the opposite though? If this fails, we know that we cleaned up, we just weren't able to remove the finalizer, so we are happy that the controller sees the exit code as 0 and removes the finalizer instead?

Reply:
There is a (maybe small but still) possibility that the pod fails to update the pod with the removed finalizer (let's say because of a resource version mismatch), exits properly but still triggers the k8s 1.20 bug with the pod exiting 137 and so the controller will not clean this, right?

Reply:
Also if we decide to keep an error log here we should exit, otherwise we'll continue with the pod finalizer update with a nil object as error does not exit.

Reply:
Does the new backoff logic change your thoughts? We now "exit" early if the GET fails, we retry 3 times total. Admittedly, it is still possible to fail all of those, exit properly, get the 137, and then we don't get cleaned up, but I dont see how to eliminate that chance, and this is still an improvement?